### PR TITLE
Pod runtimes fix

### PR DESCRIPTION
**Pull request by j-awn**
[RUNTIME]
## About the PR 

Bumped into a pair of runtime bugs when trying to fix another problem.

## Why's this needed? 

Runtimes bad!

## Changelog

N/A

### CODE REVIEW

File: code/obj/machinery.dm [hunk 1]
```diff
@@ -368,6 +368,6 @@
 	var/area/A1 = get_area(src)
 	. = ..()
 	var/area/A2 = get_area(src)
-	if(A1 != A2)
+	if(islist(A1?.machines) && islist(A2?.machines) && A1 != A2)
```
Comment (ZeWaka) [inline code/obj/machinery.dm:371]: why are you checking islist instead of just nullchecks

Comment (j-awn) [reply code/obj/machinery.dm:371]: I could probably do null-checks here instead of list-checks

File: code/obj/machinery.dm [hunk 2]
```diff
@@ -368,6 +368,6 @@
 	var/area/A1 = get_area(src)
 	. = ..()
 	var/area/A2 = get_area(src)
-	if(A1 != A2)
+	if(!isnull(A1?.machines) && !isnull(A2?.machines) && A1 != A2)
```
Comment (ZeWaka) [inline code/obj/machinery.dm:371]: fixing the symptom not the problem, .machines will never be null, by looking at references to it
However, it is valid to check if A1 and A2 exist / not null, which would be the proper fix

File: code/datums/movement_controller/pod.dm [hunk 3]
```diff
@@ -80,7 +80,7 @@
 
 
 	update_owner_dir(var/atom/movable/ship) //after move, update ddir
```
Comment (ZeWaka) [inline code/datums/movement_controller/pod.dm:82]: delete this unused argument...

File: code/datums/movement_controller/pod.dm [hunk 4]
```diff
@@ -80,7 +80,7 @@
...
-		owner.dir = last_dir
+		owner?.dir = last_dir
```
Comment (ZeWaka) [inline code/datums/movement_controller/pod.dm:83]: this is valid because if owner is null, we're in the middle of disposing the holder anyways

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/datums/movement_controller/movement_controller.dm [hunk 5]
```diff
@@ -12,4 +12,4 @@
 			SHOULD_CALL_PARENT(TRUE)
 			// stub
 
-		update_owner_dir(var/atom/movable/owner)
+		update_owner_dir()
```
Comment (ZeWaka) [inline code/datums/movement_controller/movement_controller.dm:15]: nope....

File: code/datums/movement_controller/tank.dm [hunk 6]
```diff
@@ -157,7 +157,7 @@
 		next_move = world.time + delay
 		return min(delay, next_rot-world.time)
 
-	update_owner_dir(var/atom/movable/ship) //after move, update ddir
+	update_owner_dir() //after move, update ddir
```
Comment (ZeWaka) [inline code/datums/movement_controller/tank.dm:160]: why are you touching tank controls in this PR

File: code/datums/movement_controller/movement_controller.dm (hunk 5 above)
Comment (j-awn) [reply code/datums/movement_controller/movement_controller.dm:15]: The child datum had this proc with the "extraneous argument"; this proc is only used for the tank and pod, neither of which actually use the argument. Do you think the parent datum should keep the proc argument and the child proc should remove it?

Comment (ZeWaka) [reply code/datums/movement_controller/movement_controller.dm:15]: fair, if it's not being used anywhere and there's only these two subtypes which do it different, it's ok to remove

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**